File: app/src/main/java/com/lmyby/ankihelper/data/dict/Cdepe4Database.java
package com.lmyby.ankihelper.data.dict;

import android.content.Context;

import androidx.room.Dao;
import androidx.room.Database;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.Query;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.lmyby.ankihelper.data.dict.Form; // Import the Form class

@Database(entities = {DummyEntity.class}, version = 2, exportSchema = false)
public abstract class Cdepe4Database extends RoomDatabase {

    public abstract Cdepe4Dao cdepe4Dao();

    private static Cdepe4Database instance;

    public static synchronized Cdepe4Database getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    Cdepe4Database.class, "cdepe4.db")
                    .createFromAsset("databases/cdepe4.db")
                    .fallbackToDestructiveMigration()
                    // Removed .allowMainThreadQueries() - use coroutines for async operations
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/com/lmyby/ankihelper/data/dict/Cdepe4Database.kt
package com.lmyby.ankihelper.data.dict

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

/**
 * Converted to Kotlin as part of Phase 2 data model migration
 */
@Database(entities = [DummyEntity::class], version = 2, exportSchema = false)
abstract class Cdepe4Database : RoomDatabase() {

    abstract fun cdepe4Dao(): Cdepe4Dao

    companion object {
        @Volatile
        private var instance: Cdepe4Database? = null

        @JvmStatic
        fun getInstance(context: Context): Cdepe4Database {
            return instance ?: synchronized(this) {
                instance ?: Room.databaseBuilder(
                    context.applicationContext,
                    Cdepe4Database::class.java,
                    "cdepe4.db"
                )
                    .createFromAsset("databases/cdepe4.db")
                    .fallbackToDestructiveMigration()
                    // Removed .allowMainThreadQueries() - use coroutines for async operations
                    .build().also { instance = it }
            }
        }
    }
}
